libindy/src/services/wallet/mod.rs

```diff
@@ -1511,4 +1542,39 @@ mod tests {
         let get_pool_name_res = wallet_service.get_pool_name(1);
         assert_match!(Err(WalletError::InvalidHandle(_)), get_pool_name_res);
     }
+
+    /**
+        Key rotation test
+    */
+    #[test]
+    fn wallet_service_key_rotation() {
+        _cleanup();
+
+        let wallet_service = WalletService::new();
+        wallet_service.create_wallet("pool1", "test_wallet", None, None, &_credentials()).unwrap();
+        let wallet_handle = wallet_service.open_wallet("test_wallet", None, &_credentials()).unwrap();
+
+        wallet_service.add_record(wallet_handle, "type", "key1", "value1", "{}").unwrap();
+        let record = wallet_service.get_record(wallet_handle, "type", "key1", &_fetch_options(false, false, false)).unwrap();
```

Comment:
I think it's better to fetch something and check that after key rotation fetched value is the same

Reply:
yes, that is what asserts below do.

Reply:
I see only asserts like `record.get_value().is_none()`
It's better to have something like
```
...
wallet.add_record(..., value1_const, ...)
...
assert_eq!(fetched_value, Some(value1_const));
...
rotate_keys();
...
assert_eq!(fetched_value, Some(value1_const));
```

Reply:
oh yes you are right. I brainfarted and reused wrong test. I will change.My bad